Update asynchronous_code.md (#27594)
TheOdinProject/curriculum: javascript/asynchronous_javascript_and_apis/asynchronous_code.md

1. Read this [Promises article](https://davidwalsh.name/promises). It's a good starting place and it's short and to the point.
2. Watch this [Promises video](https://youtu.be/DHvZLI7Db8E).  It's a good place to get a feel for how one might actually use promises in the wild.
3. Watch this [What is Event Loop? video](https://www.youtube.com/watch?v=8aGhZQkoFbQ) to understand how asynchronous code works in JavaScript.

> 4. Read this [Promises article](https://javascript.info/promise-basics). It's an excellent resource for beginners.

Reviewer: We need to ensure that the link text is both descriptive and unique. Since we already have a link with the text "Promises articles" let's update this one:
```suggestion
4. Read [promise basics](https://javascript.info/promise-basics), which is an excellent resource for beginners.
```

Author: I Got it @thatblindgeye

</div>

### Knowledge check

This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.

- [What is a callback?](https://developer.mozilla.org/en-US/docs/Glossary/Callback_function)
- [What is a promise?](#promises)
- [When should you use promises over callbacks?](http://callbackhell.com/)
- [What does the `.then()` function do?](https://davidwalsh.name/promises)

### Additional resources

This section contains helpful links to other content. It isn't required, so consider it supplemental.

1. The [Promises by ColorCode](https://www.youtube.com/watch?v=TnhCX0KkPqs) is a nice introduction to Promises if you need additional information.
2. For a more interactive explanation and example, try a [Scrim on asynchronous coding](https://scrimba.com/scrim/cof4e4fb797a2d0a236ea38ce?embed=odin,mini-header,no-next-up).

> 3. Read [Chapter 2: Callbacks](https://github.com/getify/You-Dont-Know-JS/blob/1st-ed/async%20%26%20performance/ch2.md) and [Chapter 3: Promises](https://github.com/getify/You-Dont-Know-JS/blob/1st-ed/async%20%26%20performance/ch3.md) from `You Don't Know JS`. In *Chapter 2*, the author explains the problems with callbacks and why callback hell will be your worst enemy (hint: it's the inversion of control and non-linear nature of callbacks). In *Chapter 3*, you go **deep** into the how and why of promises. This chapter is not the easiest read, but you'll be a promise professional if you take the time to properly digest it.  It's worth the effort.

Reviewer: Just to clean this one up a bit:

```suggestion
3. Read [Chapter 2: Callbacks](https://github.com/getify/You-Dont-Know-JS/blob/1st-ed/async%20%26%20performance/ch2.md) and [Chapter 3: Promises](https://github.com/getify/You-Dont-Know-JS/blob/1st-ed/async%20%26%20performance/ch3.md) from "You Don't Know JavaScript". Both chapters dive deep into their respective topics, which may make for a difficult - but worthwhile - read.
```

Author: I resolved it and made a pull request.
